homeassistant/components/androidtv/config_flow.py

```diff
@@ -0,0 +1,302 @@
+        """Return initial configuration form."""
+
+        apps = {APPS_NEW_ID: "Add new"}
+        apps.update(self._apps)
+        options = self.config_entry.options
+        data_schema = vol.Schema(
+            {
+                vol.Optional(CONF_APPS): vol.In(apps),
+                vol.Optional(
+                    CONF_GET_SOURCES,
+                    default=options.get(CONF_GET_SOURCES, DEFAULT_GET_SOURCES),
+                ): bool,
+                vol.Optional(
+                    CONF_EXCLUDE_UNNAMED_APPS,
+                    default=options.get(
+                        CONF_EXCLUDE_UNNAMED_APPS, DEFAULT_EXCLUDE_UNNAMED_APPS
+                    ),
+                ): bool,
+                vol.Optional(
+                    CONF_SCREENCAP,
+                    default=options.get(CONF_SCREENCAP, DEFAULT_SCREENCAP),
+                ): bool,
+                vol.Optional(
+                    CONF_TURN_OFF_COMMAND,
+                    description={
+                        "suggested_value": options.get(CONF_TURN_OFF_COMMAND, "")
+                    },
+                ): str,
+                vol.Optional(
+                    CONF_TURN_ON_COMMAND,
+                    description={
+                        "suggested_value": options.get(CONF_TURN_ON_COMMAND, "")
+                    },
+                ): str,
+                vol.Optional(
+                    CONF_STATE_DETECTION_RULES,
+                    description={
+                        "suggested_value": options.get(CONF_STATE_DETECTION_RULES, "")
+                    },
+                ): str,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="init", data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_apps(self, user_input=None, app_id=None):
+        """Handle options flow for apps list."""
+        if app_id is not None:
+            self._conf_app_id = app_id if app_id != APPS_NEW_ID else None
+            return self._async_apps_form(app_id)
+        if user_input is not None:
+            app_id = user_input.get(CONF_APP_ID, self._conf_app_id)
+            if app_id:
+                if user_input.get(CONF_APP_DELETE, False):
+                    self._apps.pop(app_id)
+                else:
+                    self._apps[app_id] = user_input.get(CONF_APP_NAME, "")
+
+        return await self.async_step_init()
+
+    @callback
+    def _async_apps_form(self, app_id):
+        """Return configuration form for apps."""
+        data_schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_APP_NAME,
+                    description={"suggested_value": self._apps.get(app_id, "")},
+                ): str,
+            }
+        )
+        if app_id == APPS_NEW_ID:
+            data_schema = data_schema.extend(
+                {
+                    vol.Optional(CONF_APP_ID): str,
+                }
+            )
```

Comment:
```suggestion
            data_schema[vol.Optional(CONF_APP_ID)] = str 
```

Reply:
If you make everything a `dict`, and then wrap it in `vol.Schema` at the end, this will work.